### PR TITLE
add beforeDestroy hook to fix blank page on reload

### PR DESCRIPTION
**Pull request by whimsicallyson**
PR Checklist:

- [ ] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Fixes issue #525

### CODE REVIEW

**Comment by ThatGalNatalie** [comment]
This fixes the blank page but when I try to view the api page on Firefox and Safari it won't load the page. In the console, it's giving the following errors: 
```
Content Security Policy: The page’s settings blocked the loading of a resource at https://unpkg.com/graphiql/graphiql.min.css (“style-src”).
```

```
Error: Loading CSS chunk interactive-api~851fd946 failed.
(/css/interactive-api~851fd946.73c442da.css)
```

**Comment by whimsicallyson** [comment]
@ThatGalNatalie this should fix Safari at least, I'm currently seeing some breaking CSS issues on Firefox that I'm gonna dig into further. Is that what you see as well?

**Comment by ThatGalNatalie** [comment]
Works for me on all the browsers. What's currently breaking?

**Comment by zhen0** [comment]
Loading fine for me!  One small visual thing - the highlighting box around the run button seems to be cut off at the bottom.  (Only seems visible in Chrome?). Not a biggie and I think is also the same in dev but I now notice it every time I click that run button. 
Screen Shot 2020-12-29 at 3 46 59 PM

File: public/index.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
       img-src * data:;
       style-src-elem 'self' 'unsafe-inline' https://fonts.googleapis.com https://kit-free.fontawesome.com https://unpkg.com;
       script-src-elem 'self' 'unsafe-inline' https://cdn.lr-ingest.io https://kit.fontawesome.com https://js.stripe.com https://unpkg.com;
-      style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://kit-free.fontawesome.com;
+      style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://kit-free.fontawesome.com https://unpkg.com;
```
Comment (znicholasbrown) [inline public/index.html:18]: Ah, taking another look at these, I think both the style and script src content CSP lines need to be more specific for graphiql

File: src/pages/InteractiveAPI/InteractiveAPI.vue [hunk 2]
```diff
@@ -93,6 +93,10 @@ query { hello }
       document.getElementById('graphiql')
     )
   },
+  beforeDestroy() {
+    const cdnScripts = ['react', 'react-dom', 'graphiql-js']
+    cdnScripts.forEach(script => document.getElementById(script).remove())
```
Comment (znicholasbrown) [inline src/pages/InteractiveAPI/InteractiveAPI.vue:98]: Is there a downside to removing these every time the component is destroyed? My understanding is that these scripts can be fairly large requests.

**Review by znicholasbrown — CHANGES_REQUESTED** [review]
Nice @whimsicallyson - this fixes the issue but I have a couple questions on the implementation.

File: src/pages/InteractiveAPI/InteractiveAPI.vue (hunk 2 above)
Comment (whimsicallyson) [reply src/pages/InteractiveAPI/InteractiveAPI.vue:98]: they don't seem comparatively huge,  235kb over network/875kb uncompressed. Also, this ensures we're getting the most recent versions (not likely an issue for navigating between pages, but could be if we were to check the cache for a file instead)

**Comment by whimsicallyson** [comment]
>  One small visual thing - the highlighting box around the run button seems to be cut off at the bottom

made a custom focus style for that @zhen0! 
Screen Shot 2020-12-30 at 2 36 05 PM

File: src/pages/InteractiveAPI/InteractiveAPI.vue (hunk 2 above)
Comment (znicholasbrown) [reply src/pages/InteractiveAPI/InteractiveAPI.vue:98]: Hm ok - that seems fine.

**Review by znicholasbrown — APPROVED** [review]
Thanks for the changes @whimsicallyson - LGTM

**Pull request merged**